src/store/State.ts

```diff
@@ -0,0 +1,45 @@
+/**
```

Comment:
I'm keeping state types in a separate file here. Maybe overly compartmentalized, but it feels useful to have a single point of access for defining state.

Reply:
I like the compartmentalized approach.